powerauth-webflow-authentication-form/src/main/java/io/getlime/security/powerauth/lib/webflow/authentication/method/form/controller/FormLoginController.java

```diff
@@ -99,7 +109,21 @@ protected AuthenticationResult authenticate(UsernamePasswordAuthenticationReques
             FormData formData = new FormDataConverter().fromOperationFormData(operation.getFormData());
             ApplicationContext applicationContext = operation.getApplicationContext();
             OperationContext operationContext = new OperationContext(operation.getOperationId(), operation.getOperationName(), operation.getOperationData(), formData, applicationContext);
-            final ObjectResponse<AuthenticationResponse> authenticateResponse = dataAdapterClient.authenticateUser(request.getUsername(), request.getPassword(), request.getOrganizationId(), operationContext);
+            final ObjectResponse<UserDetailResponse> lookupResponse = dataAdapterClient.lookupUser(request.getUsername(), request.getOrganizationId(), operationContext);
+
+            AuthenticationType authenticationType = configuration.getAuthenticationType();
+            String cipherTransformation = configuration.getCipherTransformation();
+            String passwordToSend;
+            if (authenticationType == AuthenticationType.BASIC) {
+                // Password is sent in plain text
+                passwordToSend = request.getPassword();
+            } else {
```

Comment:
I think that we should have `else-if` here (maybe rising exception if some other value is received?).